Enabled HTML formatting for flashMessages

dariobanfi opened the pull request:
As part of https://github.com/auth0/lock/issues/878#issuecomment-286882027, successfully merged through https://github.com/auth0/lock/pull/928 you enabled html formatting in error/success messages.

However this only work only through languageDictionary strings and not when the message is called through: `lock.show({flashMessage: {...})`
 
This small change should fix that.

dariobanfi commented:
(Failing tests shouldn't be because of my change I think... I just cloned the latest master)

luisrudge commented:
I triggered a rebuild. In the meantime, can you add a test for this? Thanks for the PR!

dariobanfi commented:
@luisrudge I added some component tests

luisrudge commented:
Hi @dariobanfi the tests are failing because you changed the way you render errors.
We validated [https://github.com/auth0/lock/blob/f82d025fd9015d7e381174657f38e850d6dd1e69/test/mfa_ro.ui.test.js#L85](here) that the error was shown using [this](https://github.com/auth0/lock/blob/ba69824e09329dec95a50bd98172eb2e4d0f7350/test/helper/ui.js#L195) helper function. You should make sure the tests reflect the change you made and they pass. Let me know if you need more help. Thanks again for the PR!

dariobanfi commented:
I updated the PR.
The problem with tests was caused by an inconsistency on how GlobalMessage `message` prop was passed, sometimes it was a React component (when using i18n strings) and sometimes a string (when using flashMessage).
I changed so that it's always passed as React component.

luisrudge commented:
Thanks for the PR @dariobanfi 🎉
